doc/api/stream.md

```diff
@@ -1555,17 +1555,30 @@ const cleanup = finished(rs, (err) => {
 });
 ```
 
-### `stream.pipeline(...streams, callback)`
+### `stream.pipeline(source, ...transforms, destination, callback)`
```

Comment:
Hmmm.... we usually try to have the source code argument names match that in the function definitions in the code so that (among other things) error messages make sense when users refer to the docs. (If an error says "`foo` must be a string", we want the doc to list "foo" as an argument for the relevant method.) But the relevant function signature in the source code here is `pipeline(...streams)`.  

The disparity already exists so it doesn't necessarily need to be fixed here, but I wonder if doing so might make the code easier to understand anyway. I don't think this function signature is quite correct anyway? At least looking at the code, it seems like `...transforms` is optional? Calling `pipeline()` with just source, destination, and callback works fine, right?

Assuming I'm right about all the above, I guess at a minimum, we need to make it clear that `transforms` are optional.

Reply:
The `INVALID_ARG_TYPE` and `INVALID_RETURN_TYPE` errors correctly correspond with the names here in the docs.

`...transforms` means 0 or many, i.e. `...` is 0 or many, i.e. it's optional.

The only thing here that's a bit inconsistent is the function signature in the code which might weird descriptions from linters maybe?

Reply:
Sounds good to me. Thanks.